src/jaxsim/parsers/rod/parser.py

```diff
@@ -41,30 +41,37 @@ def extract_model_data(
     Extract data from an SDF/URDF resource useful to build a JaxSim model.
 
     Args:
-        model_description: A path to an SDF/URDF file, a string containing its content,
-          or a pre-parsed/pre-built rod model.
+        model_description:
+            A path to an SDF/URDF file, a string containing its content, or
+            a pre-parsed/pre-built rod model.
         model_name: The name of the model to extract from the SDF resource.
-        is_urdf: Whether the SDF resource is a URDF file. Needed only if model_description
-            is a URDF string.
+        is_urdf:
+            Whether to force parsing the resource as a URDF file. Automatically
+            detected if not provided.
 
     Returns:
         The extracted model data.
     """
 
-    if isinstance(model_description, rod.Model):
-        sdf_model = model_description
-    else:
-        # Parse the SDF resource.
-        sdf_element = rod.Sdf.load(sdf=model_description, is_urdf=is_urdf)
-
-        if len(sdf_element.models()) == 0:
-            raise RuntimeError("Failed to find any model in SDF resource")
-
-        # Assume the SDF resource has only one model, or the desired model name is given.
-        sdf_models = {m.name: m for m in sdf_element.models()}
-        sdf_model = (
-            sdf_element.models()[0] if len(sdf_models) == 1 else sdf_models[model_name]
-        )
+    match model_description:
+        case rod.Model():
+            sdf_model = model_description
+        case rod.Sdf(None):
```

Comment:
This case was not originally part of the logic. It's a good addition having it here but there's something missing:

- Update the typing of the argument.
- Apply the logic to extract the right model used in the last case block also here (possibly with no code duplication).

Reply:
I agree with that! Addressed in https://github.com/ami-iit/jaxsim/pull/238/commits/24e91a78a0c374826827665e72277fecfce7ee41